### PR TITLE
Add team flag to get user list of a given team

### PR DESCRIPTION
**Pull request by darkLord19**
Fixes mattermost/mattermost-server#15686

### CODE REVIEW

**Comment by darkLord19** [comment]
i have added test for the feature but it is failing but not sure why as I have not worked with testing mechanism mattermsot is using so any pointers on how to figure it out are appreciated.

File: mocks/client_mock.go [hunk 1]
```diff
@@ -864,6 +864,21 @@ func (mr *MockClientMockRecorder) GetUsers(arg0, arg1, arg2 interface{}) *gomock
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "GetUsers", reflect.TypeOf((*MockClient)(nil).GetUsers), arg0, arg1, arg2)
 }
 
+// GetUsersInTeam mocks base method
+func (m *MockClient) GetUsersInTeam(arg0 string, arg1, arg2 int, arg3 string) ([]*model.User, *model.Response) {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "GetUsersInTeam", arg0, arg1, arg2)
```
Comment (streamer45) [inline mocks/client_mock.go:870]: > i have added test for the feature but it is failing but not sure why as I have not worked with testing mechanism mattermsot is using so any pointers on how to figure it out are appreciated.

This is what's causing the issue. The call is expecting 4 arguments, but only 3 given.
To fix this you can run `make mocks`, commit and push the changes.

Comment (darkLord19) [reply mocks/client_mock.go]: Thanks! Fixed it.

**Review by streamer45 — APPROVED** [review]
Thank you @darkLord19  
I've left a couple of suggestions but overall PR looks good :+1:

**Review by mgdelacroix — APPROVED** [review]
LGTM! Thanks a lot for your contribution @darkLord19!

It looks like tests are failing, would you mind taking a look?

**Review by lindy65 — APPROVED** [review]
Thanks @darkLord19 @mgdelacroix - looks good to me 👍

**Comment by hanzei** [comment]
@darkLord19 It seems like the tests are failing in CI. Could you please take a looK?

**Comment by hanzei** [comment]
Thanks for fixing the test and this great contribution @darkLord19 :+1:

**Pull request merged**